dask/utils.py

```diff
@@ -37,6 +37,61 @@ def apply(func, args, kwargs=None):
         return func(*args)
 
 
+def _deprecated(
+    *,
+    version: str = None,
+    msg: str = None,
+    use_instead: str = None,
+    category: Type[Warning] = FutureWarning,
```

Comment:
I just looked up the difference between FutureWarning and DeprecationWarning and I still don't understand when you would want to use one vs the other. Do you think it's really worth making this configurable?

Reply:
I found https://stackoverflow.com/questions/55377831/difference-between-deprecationwarning-pendingdeprecationwarning-and-futurewarni

DeprecationWarning is only going to be reported if you invoke the deprecated code from the ``__main__`` module; FutureWarning is going to be reported regardless where you use it.

I too see very little benefit in making it configurable.

Reply:
Yeah, that's a fair point. The difference has to do with the intended audience and some people also use them depending on timing (i.e. if something is currently deprecated or will be deprecated in the future). Thought I agree, this is probably an example of early generalization. I don't feel strongly whether this functionality should stay or go, so I'll remove it.

Reply:
I've found DeprecationWarnings helpful in the past when you'll have a very large number of warnings in code that's used by libraries. Users shouldn't see these warnings, since there's nothing they can do about it. Test runners like `pytest` will make DeprecationWarnings visible to the library authors when they run there tests, so they can fix the warnings before users see them. So the cycle is DeprecationWarning -> libraries fix code -> FutureWarning -> users fix code.

Reply:
Alright, thanks for the additional input @TomAugspurger. Let's keep this option then. It's two lines of code and we may find it useful.